src/v2/Components/LocationAutocompleteInput.tsx

```diff
@@ -0,0 +1,161 @@
+import React from "react"
+import { DEFAULT_MODAL_Z_INDEX, Input, InputProps } from "@artsy/palette"
+import { useLoadScript } from "v2/Utils/Hooks/useLoadScript"
+import { data as sd } from "sharify"
+import { useRef } from "react"
+import { useEffect } from "react"
+import { useState } from "react"
+import { useAppendStylesheet } from "v2/Utils/Hooks/useAppendStylesheet"
+
+const GOOGLE_PLACES_API_SRC = `https://maps.googleapis.com/maps/api/js?key=${sd.GOOGLE_MAPS_API_KEY}&libraries=places&language=en`
+
+interface LocationAutocompleteInputProps extends Omit<InputProps, "onChange"> {
+  onChange(place: Place): void
+}
+
+export const LocationAutocompleteInput: React.FC<LocationAutocompleteInputProps> = ({
+  onChange,
+  ...rest
+}) => {
+  const [ready, setReady] = useState(false)
+
+  useLoadScript({
+    id: "google-maps-js",
+    src: GOOGLE_PLACES_API_SRC,
+    onReady: () => {
+      setReady(true)
+    },
+  })
+
+  useAppendStylesheet({
+    id: "google-maps-css",
+    body: `.pac-container { z-index: ${DEFAULT_MODAL_Z_INDEX}; }`,
+  })
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const autocompleteServiceRef = useRef<any | null>(null)
+
+  useEffect(() => {
+    if (!ready || !inputRef.current) return
+
+    // @ts-ignore
+    autocompleteServiceRef.current = new google.maps.places.Autocomplete(
```

Comment:
I didn't hunt down the typings for this since it's loaded into the global async, blah blah. Probably a better way to handle this; but the component is simple enough.